### PR TITLE
pythonPackages.zc_buildout: 2.13.4 -> 3.0.0b2

### PR DESCRIPTION
**Pull request by superherointj**
Upgrade necessary because `setuptools` was upgraded to 54.2.0 and current `buildout` does not work with setuptools >52, see issue: https://github.com/buildout/buildout/issues/543

This is affecting downstream packages like [martian](https://pypi.org/project/martian/). See #136516 

---

On fixing tests:

My effort to fix `zc_buildout` tests failed. Because `zc_buildout` depends on `zope_zc_zdaemonrecipe` which depends on `zc.recipe.egg` which is broken!

7 python packages were ported in the process.
* Upstream all 6 *working* packages? [They are here in this PR.]
  * zope_bobo: init 2.4.0
  * zope_bobodoctestumentation: init 2.2.0
  * check-python-versions: init 0.18.1
  * coverage-python-version: init 0.2.0
  * types-pyyaml: init 5.4.10
  * zope_zc_recipe_deployment: init 1.3.0

* `zope_zc_zdaemonrecipe` is broken. Upstream to mark it as broken? (I wouldn't have wasted time if I knew it was broken.)

### CODE REVIEW

**Comment by lucasew** [comment]
It seems that setuptools dropped a feature buildout requires so there are two options:
- Use an older version of setuptools, before they dropped easy_install support, which seems to be 52 [1]
- Merge this PR bumping buildout to 3.x, but the problem is that this version is a prerelease and the concern is how much different would be this prerelease to the final release

I vote for the second option, setuptools has a massively wider usage, buildout is just used in a few packages [2] so if something breaks it should be trivial to fix.

Diff-wise LGTM

[1] https://github.com/buildout/buildout/pull/549
[2] https://github.com/NixOS/nixpkgs/search?q=buildout

**Comment by nixinator** [comment]
what happened to     homepage = "http://www.buildout.org";

**Comment by superherointj** [comment]
> what happened to homepage = "http://www.buildout.org";

Alphabetical order. HTTP vs HTTPS.

**Comment by hqurve** [comment]
> Alphabetical order. HTTP vs HTTPS.

Why is the homepage not https? Could you please change it to https? The https version is working (see https://www.buildout.org/en/latest/)

**Comment by superherointj** [comment]
> > Alphabetical order. HTTP vs HTTPS.
> 
> Why is the homepage not https? Could you please change it to https? The https version is working (see https://www.buildout.org/en/latest/)

It's fixed now. :-)

File: pkgs/development/python-modules/buildout/default.nix [hunk 1]
```diff
@@ -1,18 +1,43 @@
-{ lib, buildPythonPackage, fetchPypi }:
+{ lib, buildPythonPackage, fetchFromGitHub
+, setuptools, pip, wheel
+, zope_testing, manuel/*, bobo*/, zdaemon
```
Comment (hqurve) [inline pkgs/development/python-modules/buildout/default.nix:3]: here

File: pkgs/development/python-modules/buildout/default.nix [hunk 2]
```diff
@@ -1,18 +1,43 @@
...
+    wheel
+  ];
+
+  checkInputs = [
+    zope_testing
+    manuel
+    # bobo # Missing package
+    zdaemon
+    # zc_zdaemonrecipe # Missing package
+    # zc_recipe_deployment # Missing package
```
Comment (hqurve) [inline pkgs/development/python-modules/buildout/default.nix:29]: Same here. Does the default build/check fail or produce errors without these packages?

**Review by hqurve — COMMENTED** [review]
Why are some items commented out? I know they are not in nixpkgs, but why is it here?

Note: this question would help those in the future to know what to do

File: pkgs/development/python-modules/buildout/default.nix (hunk 2 above)
Comment (superherointj) [reply pkgs/development/python-modules/buildout/default.nix:29]: They are required, as you can check at: https://github.com/buildout/buildout/blob/master/setup.py#L57

File: pkgs/development/python-modules/buildout/default.nix (hunk 1 above)
Comment (superherointj) [reply pkgs/development/python-modules/buildout/default.nix:3]: It was a placeholder for the `bobo` python package. I can remove it.

File: pkgs/development/python-modules/buildout/default.nix (hunk 2 above)
Comment (superherointj) [reply pkgs/development/python-modules/buildout/default.nix:29]: I haven't fixed tests. I suppose the lack of dependencies will block tests. But I'm not familiar with this test suite. I'm only trying to bump version to fix another package that got out of sync with another bumped dependency (setuptools).

File: pkgs/development/python-modules/buildout/default.nix (hunk 1 above)
Comment (superherointj) [reply pkgs/development/python-modules/buildout/default.nix]: Removed.

File: pkgs/development/python-modules/buildout/default.nix (hunk 2 above)
Comment (superherointj) [reply pkgs/development/python-modules/buildout/default.nix:29]: These dependencies are listed as required for tests in `setup.py`. https://github.com/buildout/buildout/blob/master/setup.py#L58

**Pull request closed without merging**